packages/stargate/src/aminotypes.ts

```diff
@@ -1,61 +1,381 @@
+        assert(description.identity, "missing description.identity");
+        assert(description.website, "missing description.website");
+        assert(description.securityContact, "missing description.securityContact");
+        assert(description.details, "missing description.details");
+        assert(commission, "missing commission");
+        assert(commission.rate, "missing commission.rate");
+        assert(commission.maxRate, "missing commission.maxRate");
+        assert(commission.maxChangeRate, "missing commission.maxChangeRate");
+        assert(minSelfDelegation, "missing minSelfDelegation");
+        assert(delegatorAddress, "missing delegatorAddress");
+        assert(validatorAddress, "missing validatorAddress");
+        assert(pubkey, "missing pubkey");
+        assert(pubkey.value, "missing pubkey.value");
+        assert(value, "missing value");
+        assert(value.amount, "missing value.amount");
+        assert(value.denom, "missing value.denom");
+        return {
+          description: {
+            moniker: description.moniker,
+            identity: description.identity,
+            website: description.website,
+            security_contact: description.securityContact,
+            details: description.details,
+          },
+          commission: {
+            rate: commission.rate,
+            max_rate: commission.maxRate,
+            max_change_rate: commission.maxChangeRate,
+          },
+          min_self_delegation: minSelfDelegation,
+          delegator_address: delegatorAddress,
+          validator_address: validatorAddress,
+          pubkey: encodeBech32Pubkey(
+            {
+              type: "tendermint/PubKeySecp256k1",
+              value: toBase64(pubkey.value),
+            },
+            prefix,
+          ),
+          value: {
+            amount: value.amount,
+            denom: value.denom,
+          },
+        };
+      },
+      fromAmino: ({
+        description,
+        commission,
+        min_self_delegation,
+        delegator_address,
+        validator_address,
+        pubkey,
+        value,
+      }: MsgCreateValidator["value"]): IMsgCreateValidator => ({
+        description: {
+          moniker: description.moniker,
+          identity: description.identity,
+          website: description.website,
+          securityContact: description.security_contact,
+          details: description.details,
+        },
+        commission: {
+          rate: commission.rate,
+          maxRate: commission.max_rate,
+          maxChangeRate: commission.max_change_rate,
+        },
+        minSelfDelegation: min_self_delegation,
+        delegatorAddress: delegator_address,
+        validatorAddress: validator_address,
+        pubkey: {
+          // TODO: How do we detect/handle other kinds of PubKey?
+          type_url: "/cosmos.crypto.secp256k1.PubKey",
+          value: fromBase64(decodeBech32Pubkey(pubkey).value),
```

Comment:
The pubkey type is encoded in `pubkey` and available in `ecodeBech32Pubkey(pubkey).type`. I'd switch over that and throw an error for everything that is not `tendermint/PubKeySecp256k1`. This makes it safe and easy to extend if needed.